### PR TITLE
Transform sparse_coding_test into example.

### PR DESCRIPTION
**Pull request by Algue-Rythme**
The file `sparse_coding_test` was slow because it was running three tests sequentially with different hyper-parameters, while the file `sparse_coding` was a library not running anything.  
  
To stay coherent with others examples I suggest to remove the file `sparse_coding_test` and move its content in the `main` function of `sparse_coding`. Instead of running three tests sequentially, only one is run by default.  
  
Various hyper-parameters have been moved in `FLAGS` at the top of the file.  
  
I added some prints.

### CODE REVIEW

**Comment by mblondel** [comment]
I suggest we proceed with merging this PR as these tests are really slowing down CI. @Algue-Rythme, can you check that the PR is ready for merge on your side? Thanks!

File: examples/implicit_diff/sparse_coding.py [hunk 1]
```diff
@@ -17,19 +17,41 @@
...
+
+
+flags.DEFINE_integer("num_examples", 74, "NUmber of examples.")
+flags.DEFINE_integer("num_components", 7, "Number of atoms in dictionnary.")
+flags.DEFINE_integer("num_features", 13, "Number of features.")
+flags.DEFINE_integer("sparse_coding_maxiter", 100, "Number of iterations for sparse coding.")
+flags.DEFINE_integer("maxiter", 10, "Number of iterations of the outer loop.")
+flags.DEFINE_float("elastic_penalty", 0.01, "Strength of L2 penalty relative to L1.")
+flags.DEFINE_float("regularization", 0.01, "Regularization strength of elastic penalty.")
+flags.DEFINE_enum("reconstruction_loss", "none", ["none", "abs", "hubert"], "Loss used to build dictionnary.")
```
Comment (mblondel) [inline examples/implicit_diff/sparse_coding.py:52]: Replace "none" with "squared".

**Comment by Algue-Rythme** [comment]
Ready for merge.

**Review by mblondel — APPROVED** [review]
LGTM!

File: examples/implicit_diff/sparse_coding.py [hunk 2]
```diff
@@ -17,19 +17,41 @@
...
+
+
+flags.DEFINE_integer("num_examples", 74, "NUmber of examples.")
+flags.DEFINE_integer("num_components", 7, "Number of atoms in dictionnary.")
+flags.DEFINE_integer("num_features", 13, "Number of features.")
+flags.DEFINE_integer("sparse_coding_maxiter", 100, "Number of iterations for sparse coding.")
+flags.DEFINE_integer("maxiter", 10, "Number of iterations of the outer loop.")
+flags.DEFINE_float("elastic_penalty", 0.01, "Strength of L2 penalty relative to L1.")
+flags.DEFINE_float("regularization", 0.01, "Regularization strength of elastic penalty.")
+flags.DEFINE_enum("reconstruction_loss", "squared", ["squared", "abs", "hubert"], "Loss used to build dictionnary.")
```
Comment (mblondel) [inline examples/implicit_diff/sparse_coding.py:52]: La loss de Hubert !

**Pull request merged**